### PR TITLE
add status to project (current/archived)

### PR DESCRIPTION
**Pull request by SimonLab**
see #142
status-project

### CODE REVIEW

File: web/controllers/redcap/http_client.ex [hunk 1]
```diff
@@ -55,32 +55,33 @@ defmodule ResearchResource.Redcap.HTTPClient do
...
   @doc """
   Filter and convert a list of Redcap fields:
   [%{name: "project name", description: "description of the project"}, ...]
  """
-  defp filter_projects({:ok, res}) do
-    {:ok, data} = Poison.Parser.parse(res.body)
+  defp filter_projects(data) do
     data
-    |> Enum.filter(fn(question) -> question["field_label"] == "name" || question["field_label"] == "description" end)
+    |> Enum.filter(fn(question) -> Enum.member?(~w(name description status), question["field_label"]) end)
```
Comment (samhstn) [inline web/controllers/redcap/http_client.ex]: Nice 👍

File: web/controllers/redcap/http_client.ex [hunk 2]
```diff
@@ -55,32 +55,33 @@ defmodule ResearchResource.Redcap.HTTPClient do
...
   end
 
   defp info_project({_p, values}) do
     Enum.reduce(values, %{}, fn(field, acc) ->
-      case field["field_label"] do
-        "name" -> Map.put(acc, :name, field["field_annotation"])
-        "description" -> Map.put(acc, :description,field["field_annotation"])
-      end
+      Map.put(acc, String.to_atom(field["field_label"]), field["field_annotation"])
     end)
```
Comment (samhstn) [inline web/controllers/redcap/http_client.ex]: This is much nicer, but we can do it without a reduce:
```elixir
defp info_project({_p, values}) do
  Map.new(values, fn %{"field_label" => label, "field_annotation" => annotation} ->
    {String.to_atom(label), annotation}
  end)
end
```

Or something like that

**Review by samhstn — COMMENTED** [review]
Looks good @SimonLab

**Review by samhstn — APPROVED** [review]
Looks good @SimonLab

**Review by Danwhy — APPROVED** [review]
:+1:

**Pull request merged**